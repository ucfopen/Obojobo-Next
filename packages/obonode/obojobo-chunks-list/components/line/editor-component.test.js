import React from 'react'
import renderer from 'react-test-renderer'

import Line from './editor-component'

describe('Line Editor Node', () => {
	test('renders as expected', () => {
		const component = renderer.create(
			<Line
				element={{
					content: { hangingIndent: true }
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('renders without content', () => {
		const component = renderer.create(<Line element={{}} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('renders without hanging indent', () => {
		const component = renderer.create(
			<Line
				element={{
					content: {}
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('renders with children', () => {
		const component = renderer.create(
			<Line
				element={{
					content: {}
				}}
			>
				children!
			</Line>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('renders with children with colored text', () => {
		const coloredTextProps = {
			node: {
				children: [
					{
						color: 'Black'
					}
				]
			}
		}

		const coloredTextChild = React.createElement('div', coloredTextProps)
		const line = React.createElement(Line, { element: { content: {} } }, coloredTextChild)

		const component = renderer.create(line)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})
})
