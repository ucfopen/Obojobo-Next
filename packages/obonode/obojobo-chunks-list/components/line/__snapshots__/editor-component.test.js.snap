// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line Editor Node renders as expected 1`] = `
<div>
  <li
    data-hanging-indent={true}
  />
</div>
`;

exports[`Line Editor Node renders with children 1`] = `
<div>
  <li>
    children!
  </li>
</div>
`;

exports[`Line Editor Node renders with children with colored text 1`] = `
<div>
  <li
    style={
      Object {
        "color": "Black",
      }
    }
  >
    <div
      node={
        Object {
          "children": Array [
            Object {
              "color": "Black",
            },
          ],
        }
      }
    />
  </li>
</div>
`;

exports[`Line Editor Node renders without content 1`] = `
<div>
  <li />
</div>
`;

exports[`Line Editor Node renders without hanging indent 1`] = `
<div>
  <li />
</div>
`;
